lib/pwwka/receiver.rb

```diff
@@ -47,18 +40,23 @@ def self.subscribe(handler_klass, queue_name,
       rescue Interrupt => _
         # TODO: trap TERM within channel.work_pool
         info "Interrupting queue #{queue_name} subscriber safely"
-      ensure
-        receiver.channel_connector.connection_close
       end
       return receiver
     end
 
-    def topic_queue
-      @topic_queue ||= begin
+    def subscribe(**kwargs, &block)
+      connect do |connection|
+        @channel = connection.channel
+        @topic_exchange = connection.topic_exchange
```

Comment:
is it necessary to set these instance variables here?  this seems risky and error prone

Reply:
The values aren't available before `connect` is called, and it's necessary to keep this state to conform to the existing interface of this class. The `ensure` below should be clearing out `@topic_exchange` too, though.